metafield.go

```diff
@@ -37,20 +37,72 @@ type MetafieldServiceOp struct {
 	resourceID int64
 }
 
+type metafieldType string
+
+// https://shopify.dev/docs/api/admin-rest/2023-07/resources/metafield#resource-object
+const (
+	//True or false.
+	MetafieldTypeBoolean metafieldType = "boolean"
+
+	//A hexidecimal color code, #fff123.
+	MetafieldTypeColor metafieldType = "color" //#fff123
+
+	//ISO601, YYYY-MM-DD.
+	MetafieldTypeDate metafieldType = "date"
+
+	//ISO8601, YYYY-MM-DDTHH:MM:SS.
+	MetafieldTypeDatetime metafieldType = "date_time"
+
+	//JSON, {"value:" 25.0, "unit": "cm"}.
+	MetafieldTypeDimension metafieldType = "dimension"
+
+	//{"ingredient": "flour", "amount": 0.3}.
+	MetafieldTypeJSON metafieldType = "json"
+
+	//JSON, {"amount": 5.99, "currency_code": "CAD"}.
+	MetafieldTypeMoney metafieldType = "money"
+
+	//lines of text separated with newline characters.
+	MetafieldTypeMultiLineTextField metafieldType = "multi_line_text_field"
+
+	//10.4.
+	MetafieldTypeNumberDecimal metafieldType = "number_decimal"
+
+	//10.
+	MetafieldTypeNumberInteger metafieldType = "number_integer"
+
+	//JSON, {"value": "3.5", "scale_min": "1.0", "scale_max": "5.0"}.
+	MetafieldTypeRating metafieldType = "rating"
+
+	//JSON, {"type": "root","children": [{"type": "paragraph","children": [{"type": "text","value": "Bold text.","bold": true}]}]}.
+	MetafieldTypeRichTextField metafieldType = "rich_text_field"
+
+	//A single line of text. Do not use for numbers or dates, use correct type instead!
+	MetafieldTypeSingleLineTextField metafieldType = "MetafieldTypeSingleLineTextField_text_field"
```

Comment:
Something is wrong here (and everywhere else where you use `MetafieldTypeSingleLineTextField_text_field`) - this needs to be `single_line_text_field`

Reply:
I fixed this.  Looks like I did some half-baked copy-and-replace error.